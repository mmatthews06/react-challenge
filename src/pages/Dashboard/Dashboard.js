import React, { useState } from 'react';
import Grid from '@material-ui/core/Grid';
import StatusCounts from '../../components/StatusCounts/StatusCounts';
import UploadButton from '../../components/UploadButton/UploadButton';

export default function Dashboard() {
  const [samples, setSamples] = useState([]);
  const activeCount = samples.filter((s) => s.State === 'active').length;
  const queuedCount = samples.filter((s) => s.State === 'queued').length;
  const onHoldCount = samples.filter((s) => s.State === 'hold').length;

  return (
    <Grid container direction="column" spacing={2} justify="center">
      <Grid item>
        <h2>Dashboard</h2>
      </Grid>
      <Grid container item spacing={5} justify="center" alignItems="center">
        <Grid item>
          <StatusCounts
            active={activeCount}
            queued={queuedCount}
            onHold={onHoldCount}
          />
        </Grid>
        <Grid item>
          <UploadButton setSamples={setSamples} />
        </Grid>
      </Grid>
      <Grid item>
        <hr style={{ height: '2px', backgroundColor: 'blue', color: 'blue', width: '90%' }} />
      </Grid>
    </Grid>
  );
}
