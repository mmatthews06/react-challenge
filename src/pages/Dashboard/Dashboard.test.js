/* global describe, expect, it */
import React from 'react';
import { shallow } from 'enzyme';

import Dashboard from './Dashboard';

describe('Dashboard', () => {
  it('should match a snapshot', () => {
    expect(shallow(<Dashboard />)).toMatchSnapshot();
  });
});
