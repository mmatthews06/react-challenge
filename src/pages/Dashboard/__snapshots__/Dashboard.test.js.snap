// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should match a snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="column"
  justify="center"
  spacing={2}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <h2>
      Dashboard
    </h2>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    item={true}
    justify="center"
    spacing={5}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <StatusCounts
        active={0}
        onHold={0}
        queued={0}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <UploadButton
        setSamples={[Function]}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <hr
      style={
        Object {
          "backgroundColor": "blue",
          "color": "blue",
          "height": "2px",
          "width": "90%",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
